cassiopeia/tools/lineage_simulator.py

```diff
@@ -0,0 +1,108 @@
+import abc
+from typing import List
+
+import networkx as nx
+
+from .tree import Tree
+
+
+class LineageSimulator(abc.ABC):
+    r"""
+    Abstract base class for lineage simulators.
+
+    A LineageSimulator implements the method simulate_lineage that generates a
+    lineage tree (i.e. a phylogeny, in the form of a Tree).
+    """
+
+    @abc.abstractmethod
+    def simulate_lineage(self) -> Tree:
+        r"""
+        Simulates a lineage tree, i.e. a Tree with branch lengths and age
+        specified for each node. Additional information such as cell fitness,
+        etc. might be specified by more complex simulators.
+        """
+
+
+class PerfectBinaryTree(LineageSimulator):
+    r"""
+    Generates a perfect binary tree with given branch lengths at each depth.
+
+    Args:
+        generation_branch_lengths: The branches at depth d in the tree will have
+            length generation_branch_lengths[d]
+    """
+
+    def __init__(self, generation_branch_lengths: List[float]):
+        self.generation_branch_lengths = generation_branch_lengths[:]
+
+    def simulate_lineage(self) -> Tree:
+        r"""
+        See base class.
+        """
+        generation_branch_lengths = self.generation_branch_lengths
+        n_generations = len(generation_branch_lengths)
+        tree = nx.DiGraph()
+        tree.add_nodes_from(range(2 ** (n_generations + 1) - 1))
+        edges = [
+            (int((child - 1) / 2), child)
+            for child in range(1, 2 ** (n_generations + 1) - 1)
+        ]
```

Comment:
I think you can make use of `nx.balanced_tree(2, number_of_generations, create_using=nx.DiGraph)`

Reply:
Didn't know, will look at this, thanks!